pooch/tests/test_processors.py

```diff
@@ -255,3 +255,28 @@ def _unpacking_expected_paths_and_logs(archive, members, path, name):
             log_lines.append(f"Extracting '{member}'")
         true_paths = set(true_paths)
     return true_paths, log_lines
+
+
+@pytest.mark.network
+@pytest.mark.parametrize(
+    "processor_class,extension",
+    [(Unzip, ".zip"), (Untar, ".tar.gz")],
+)
+def test_unpacking_members_then_no_members(processor_class, extension):
+    """Test that calling with members then without them works.
+
+    https://github.com/fatiando/pooch/issues/364
+    """
+
+    with TemporaryDirectory() as local_store:
+        pup = Pooch(path=Path(local_store), base_url=BASEURL, registry=REGISTRY)
+
+        # Do a first fetch with incorrect member
+        processor1 = processor_class(members=["i don't exist"])
+        filenames1 = pup.fetch("store" + extension, processor=processor1)
+        assert len(filenames1) == 0
```

Comment:
```suggestion
        processor1 = processor_class(members=["tiny-data.txt"])
        filenames1 = pup.fetch("store" + extension, processor=processor1)
        assert len(filenames1) == 1
```

@jni wouldn't it make more sense to test extracting a single file that does exist? This should still cause a problem with the old behavior right? Plus, using a bad file name should really cause an error (as you mentioned previously, probably best to open a separate issue 🙂).